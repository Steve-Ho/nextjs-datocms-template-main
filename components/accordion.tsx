import {useState, useContext, useCallback, useMemo, createContext} from 'react';
import type * as React from 'react';
import {isNil, reject, equals} from 'remeda';
import {ChevronUpIcon, ChevronDownIcon} from '@heroicons/react/solid';

type ContainerContext = {
	toggleActivePanel?: ((label: string) => void);
	activePanels: string[];
};

type ExpandMode = 'multiple' | 'single';

type ContainerProperties = {
	expandMode?: ExpandMode;
	children: JSX.Element[];
};

type PanelProperties = {
	children: JSX.Element | JSX.Element[];
	label: string;
	icon?: React.ElementType;
};

const containerContext = createContext<ContainerContext>({
	activePanels: [],
});

const Container = ({children, expandMode = 'single'}: ContainerProperties): JSX.Element => {
	const [activePanels, setActivePanels] = useState<string[]>([]);

	const toggleActivePanel = useCallback((label: string) => {
		if (expandMode === 'single') {
			setActivePanels(
				activePanels.includes(label)
					? []
					: [label],
			);
		}

		if (expandMode === 'multiple') {
			setActivePanels(
				activePanels.includes(label)
					? reject(activePanels, equals(label))
					: [...activePanels, label],
			);
		}
	}, [activePanels, expandMode]);

	const context = useMemo(() => ({
		toggleActivePanel,
		activePanels,
	}), [toggleActivePanel, activePanels]);

	return (
		<section className="space-y-3">
			<containerContext.Provider value={context}>
				{children}
			</containerContext.Provider>
		</section>
	);
};

const generatePanelButtonClasses = (isActive: boolean): string => `
	flex
	w-full
	items-center
	space-x-2
	flex-grow
	text-left
	subheading
	py-2
	border-b-2
	${isActive ? 'border-true-gray-400' : 'border-true-gray-200'}
	${isActive ? '' : 'text-true-gray-400'}
`;

const Panel = ({children, label, icon: Icon}: PanelProperties): JSX.Element => {
	const {toggleActivePanel, activePanels} = useContext(containerContext);

	const handleTogglePanel = useCallback(() => {
		toggleActivePanel?.(label);
	}, [toggleActivePanel, label]);

	const isActive = useMemo(() => activePanels.includes(label), [activePanels, label]);

	const classes = useMemo(() => ({
		button: generatePanelButtonClasses(isActive),
	}), [isActive]);

	return (
		<article>
			<header>
				<button key={label} className={classes.button} type="button" onClick={handleTogglePanel}>
					{!isNil(Icon) && <Icon className="w-5"/>}
					<span>{label}</span>
					{isActive && <ChevronUpIcon className="!ml-auto w-5"/>}
					{!isActive && <ChevronDownIcon className="!ml-auto w-5"/>}
				</button>
			</header>
			<div className="mt-4">
				{isActive && children}
			</div>
		</article>
	);
};

const Accordion = {
	Container,
	Panel,
};

export default Accordion;

export type {
	ContainerContext,
	ExpandMode,
	ContainerProperties as ContainerProps,
	PanelProperties as PanelProps,
};
