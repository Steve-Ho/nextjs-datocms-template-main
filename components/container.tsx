import {useMemo} from 'react';
import {collapse} from '@growthops/ext-ts';

type ContainerProperties = {
	children: Array<JSX.Element | boolean> | JSX.Element | boolean;
	className?: string;
};

const Container = ({className = '', children}: ContainerProperties): JSX.Element => {
	const classes = useMemo(() => ({
		root: collapse(['container', className]),
	}), [className]);

	return (
		<div className={classes.root}>
			{children}
		</div>
	);
};

export default Container;

export type {
	ContainerProperties as ContainerProps,
};
