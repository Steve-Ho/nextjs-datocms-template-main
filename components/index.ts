export {default as Accordion} from './accordion';
export {default as Button} from './button';
export {default as Card} from './card';
export {default as Checkbox} from './checkbox';
export {default as Container} from './container';
export {default as Heading} from './heading';
export {default as Input} from './input';
export {default as PreviewMode} from './preview-mode';
export {default as ProgressBar} from './progress-bar';
export {default as Radio} from './radio';
export {default as Region} from './region';
export {default as Select} from './select';
export {default as Tab} from './tab';
export {default as Text} from './text';
export {default as Textarea} from './textarea';
