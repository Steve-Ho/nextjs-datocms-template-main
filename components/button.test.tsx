import {expect, test} from '@jest/globals';
import {create} from 'react-test-renderer';
import {CheckCircleIcon} from '@heroicons/react/solid';
import {Button} from '@app/components';
import type {Variant, Size, IconAlignment} from './button';

const variants: Variant[] = ['primary', 'secondary', 'tertiary'];
const sizes: Size[] = ['small', 'regular', 'large'];
const iconAlignments: IconAlignment[] = ['left', 'right'];

test('snapshots — semantic', () => {
	for (const variant of variants) {
		for (const size of sizes) {
			for (const iconAlignment of iconAlignments) {
				const result = create(
					<Button.Semantic
						label="Click me"
						variant={variant}
						size={size}
						icon={{svg: CheckCircleIcon, alignment: iconAlignment}}
					/>,
				).toJSON();

				expect(result).toMatchSnapshot();
			}
		}
	}
});

test('snapshots — link', () => {
	for (const variant of variants) {
		for (const size of sizes) {
			for (const iconAlignment of iconAlignments) {
				const result = create(
					<Button.Link
						label="Click me"
						variant={variant}
						size={size}
						icon={{svg: CheckCircleIcon, alignment: iconAlignment}}
					/>,
				).toJSON();

				expect(result).toMatchSnapshot();
			}
		}
	}
});
