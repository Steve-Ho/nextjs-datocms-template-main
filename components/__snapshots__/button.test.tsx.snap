// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots — link 1`] = `
<a
  className="inline-flex items-center rounded !leading-none bg-true-gray-600 text-white px-5 py-3 text-small space-x-2"
  rel="noreferrer"
>
  <svg
    aria-hidden="true"
    className="w-4 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</a>
`;

exports[`snapshots — link 2`] = `
<a
  className="inline-flex items-center rounded !leading-none bg-true-gray-600 text-white px-5 py-3 text-small space-x-2"
  rel="noreferrer"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-4 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</a>
`;

exports[`snapshots — link 3`] = `
<a
  className="inline-flex items-center rounded !leading-none bg-true-gray-600 text-white px-6 py-4 heading-six space-x-3"
  rel="noreferrer"
>
  <svg
    aria-hidden="true"
    className="w-5 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</a>
`;

exports[`snapshots — link 4`] = `
<a
  className="inline-flex items-center rounded !leading-none bg-true-gray-600 text-white px-6 py-4 heading-six space-x-3"
  rel="noreferrer"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-5 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</a>
`;

exports[`snapshots — link 5`] = `
<a
  className="inline-flex items-center rounded !leading-none bg-true-gray-600 text-white px-8 py-5 heading-four space-x-4"
  rel="noreferrer"
>
  <svg
    aria-hidden="true"
    className="w-7 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</a>
`;

exports[`snapshots — link 6`] = `
<a
  className="inline-flex items-center rounded !leading-none bg-true-gray-600 text-white px-8 py-5 heading-four space-x-4"
  rel="noreferrer"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-7 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</a>
`;

exports[`snapshots — link 7`] = `
<a
  className="inline-flex items-center rounded !leading-none bg-true-gray-300 text-true-gray-600 px-5 py-3 text-small space-x-2"
  rel="noreferrer"
>
  <svg
    aria-hidden="true"
    className="w-4 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</a>
`;

exports[`snapshots — link 8`] = `
<a
  className="inline-flex items-center rounded !leading-none bg-true-gray-300 text-true-gray-600 px-5 py-3 text-small space-x-2"
  rel="noreferrer"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-4 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</a>
`;

exports[`snapshots — link 9`] = `
<a
  className="inline-flex items-center rounded !leading-none bg-true-gray-300 text-true-gray-600 px-6 py-4 heading-six space-x-3"
  rel="noreferrer"
>
  <svg
    aria-hidden="true"
    className="w-5 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</a>
`;

exports[`snapshots — link 10`] = `
<a
  className="inline-flex items-center rounded !leading-none bg-true-gray-300 text-true-gray-600 px-6 py-4 heading-six space-x-3"
  rel="noreferrer"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-5 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</a>
`;

exports[`snapshots — link 11`] = `
<a
  className="inline-flex items-center rounded !leading-none bg-true-gray-300 text-true-gray-600 px-8 py-5 heading-four space-x-4"
  rel="noreferrer"
>
  <svg
    aria-hidden="true"
    className="w-7 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</a>
`;

exports[`snapshots — link 12`] = `
<a
  className="inline-flex items-center rounded !leading-none bg-true-gray-300 text-true-gray-600 px-8 py-5 heading-four space-x-4"
  rel="noreferrer"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-7 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</a>
`;

exports[`snapshots — link 13`] = `
<a
  className="inline-flex items-center rounded !leading-none border border-true-gray-300 text-true-gray-700 px-5 py-3 text-small space-x-2"
  rel="noreferrer"
>
  <svg
    aria-hidden="true"
    className="w-4 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</a>
`;

exports[`snapshots — link 14`] = `
<a
  className="inline-flex items-center rounded !leading-none border border-true-gray-300 text-true-gray-700 px-5 py-3 text-small space-x-2"
  rel="noreferrer"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-4 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</a>
`;

exports[`snapshots — link 15`] = `
<a
  className="inline-flex items-center rounded !leading-none border border-true-gray-300 text-true-gray-700 px-6 py-4 heading-six space-x-3"
  rel="noreferrer"
>
  <svg
    aria-hidden="true"
    className="w-5 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</a>
`;

exports[`snapshots — link 16`] = `
<a
  className="inline-flex items-center rounded !leading-none border border-true-gray-300 text-true-gray-700 px-6 py-4 heading-six space-x-3"
  rel="noreferrer"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-5 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</a>
`;

exports[`snapshots — link 17`] = `
<a
  className="inline-flex items-center rounded !leading-none border border-true-gray-300 text-true-gray-700 px-8 py-5 heading-four space-x-4"
  rel="noreferrer"
>
  <svg
    aria-hidden="true"
    className="w-7 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</a>
`;

exports[`snapshots — link 18`] = `
<a
  className="inline-flex items-center rounded !leading-none border border-true-gray-300 text-true-gray-700 px-8 py-5 heading-four space-x-4"
  rel="noreferrer"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-7 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</a>
`;

exports[`snapshots — semantic 1`] = `
<button
  className="inline-flex items-center rounded !leading-none bg-true-gray-600 text-white px-5 py-3 text-small space-x-2"
  type="button"
>
  <svg
    aria-hidden="true"
    className="w-4 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</button>
`;

exports[`snapshots — semantic 2`] = `
<button
  className="inline-flex items-center rounded !leading-none bg-true-gray-600 text-white px-5 py-3 text-small space-x-2"
  type="button"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-4 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`snapshots — semantic 3`] = `
<button
  className="inline-flex items-center rounded !leading-none bg-true-gray-600 text-white px-6 py-4 heading-six space-x-3"
  type="button"
>
  <svg
    aria-hidden="true"
    className="w-5 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</button>
`;

exports[`snapshots — semantic 4`] = `
<button
  className="inline-flex items-center rounded !leading-none bg-true-gray-600 text-white px-6 py-4 heading-six space-x-3"
  type="button"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-5 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`snapshots — semantic 5`] = `
<button
  className="inline-flex items-center rounded !leading-none bg-true-gray-600 text-white px-8 py-5 heading-four space-x-4"
  type="button"
>
  <svg
    aria-hidden="true"
    className="w-7 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</button>
`;

exports[`snapshots — semantic 6`] = `
<button
  className="inline-flex items-center rounded !leading-none bg-true-gray-600 text-white px-8 py-5 heading-four space-x-4"
  type="button"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-7 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`snapshots — semantic 7`] = `
<button
  className="inline-flex items-center rounded !leading-none bg-true-gray-300 text-true-gray-600 px-5 py-3 text-small space-x-2"
  type="button"
>
  <svg
    aria-hidden="true"
    className="w-4 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</button>
`;

exports[`snapshots — semantic 8`] = `
<button
  className="inline-flex items-center rounded !leading-none bg-true-gray-300 text-true-gray-600 px-5 py-3 text-small space-x-2"
  type="button"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-4 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`snapshots — semantic 9`] = `
<button
  className="inline-flex items-center rounded !leading-none bg-true-gray-300 text-true-gray-600 px-6 py-4 heading-six space-x-3"
  type="button"
>
  <svg
    aria-hidden="true"
    className="w-5 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</button>
`;

exports[`snapshots — semantic 10`] = `
<button
  className="inline-flex items-center rounded !leading-none bg-true-gray-300 text-true-gray-600 px-6 py-4 heading-six space-x-3"
  type="button"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-5 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`snapshots — semantic 11`] = `
<button
  className="inline-flex items-center rounded !leading-none bg-true-gray-300 text-true-gray-600 px-8 py-5 heading-four space-x-4"
  type="button"
>
  <svg
    aria-hidden="true"
    className="w-7 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</button>
`;

exports[`snapshots — semantic 12`] = `
<button
  className="inline-flex items-center rounded !leading-none bg-true-gray-300 text-true-gray-600 px-8 py-5 heading-four space-x-4"
  type="button"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-7 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`snapshots — semantic 13`] = `
<button
  className="inline-flex items-center rounded !leading-none border border-true-gray-300 text-true-gray-700 px-5 py-3 text-small space-x-2"
  type="button"
>
  <svg
    aria-hidden="true"
    className="w-4 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</button>
`;

exports[`snapshots — semantic 14`] = `
<button
  className="inline-flex items-center rounded !leading-none border border-true-gray-300 text-true-gray-700 px-5 py-3 text-small space-x-2"
  type="button"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-4 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`snapshots — semantic 15`] = `
<button
  className="inline-flex items-center rounded !leading-none border border-true-gray-300 text-true-gray-700 px-6 py-4 heading-six space-x-3"
  type="button"
>
  <svg
    aria-hidden="true"
    className="w-5 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</button>
`;

exports[`snapshots — semantic 16`] = `
<button
  className="inline-flex items-center rounded !leading-none border border-true-gray-300 text-true-gray-700 px-6 py-4 heading-six space-x-3"
  type="button"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-5 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`snapshots — semantic 17`] = `
<button
  className="inline-flex items-center rounded !leading-none border border-true-gray-300 text-true-gray-700 px-8 py-5 heading-four space-x-4"
  type="button"
>
  <svg
    aria-hidden="true"
    className="w-7 -ml-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
  <span>
    Click me
  </span>
</button>
`;

exports[`snapshots — semantic 18`] = `
<button
  className="inline-flex items-center rounded !leading-none border border-true-gray-300 text-true-gray-700 px-8 py-5 heading-four space-x-4"
  type="button"
>
  <span>
    Click me
  </span>
  <svg
    aria-hidden="true"
    className="w-7 !-mr-2"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;
